Objects/frameobject.c

```diff
@@ -1319,6 +1301,16 @@ _PyFrame_LocalsToFast(_PyInterpreterFrame *frame, int clear)
             }
         }
         else if (value != oldvalue) {
+            if (value == NULL) {
+                // Probably can't delete this, since the compiler's flow
+                // analysis may have already "proven" that it exists here:
+                const char *e = "assigning None to unbound local %R";
+                if (PyErr_WarnFormat(PyExc_RuntimeWarning, 0, e, name)) {
+                    // It's okay if frame_obj is NULL, just try anyways:
+                    PyErr_WriteUnraisable((PyObject *)frame->frame_obj);
+                }
+                value = Py_NewRef(Py_None);
```

Comment:
This leaks I think, as `value` gets incref'd below.

Reply:
I don't think it does. `value` is normally a new reference returned from `PyObject_GetItem`, and line 1317 does `Py_XDECREF(value)` for all branches.

Reply:
:+1:
I'd missed the decref on line 1317.